packages/11ty/content/_assets/javascript/application/canvas-panel.js

```diff
@@ -1,51 +1,107 @@
+      const disabledInput = figure.querySelector('[disabled]')
+      disabledInput ? disabledInput.removeAttribute('disabled') : null
+    }
+  }
+
+  const toggleAnnotation = ({ checked }) => {
+    canvasPanel.applyStyles(id, { opacity: Number(!!checked) })
+  }
+
+  switch (annotationType) {
+    case 'choice':
+      toggleChoice(element)
+      break
+    case 'annotation':
+      toggleAnnotation(element)
+      break
+    default:
+      console.warn(`Invalid annotation type ${annotationType}`)
+      break
+  }
+}
+
+/**
+ * Scroll to a figure, select annotations and/or region, and update the URL
+ * @param  {String} figureId    The id of the figure in figures.yaml
+ * @param  {Array} annotationIds  The IIIF ids of the annotations to select
+ * @param  {String} region      The canvas region
+ */
+const goToCanvasState = function (figureId, annotationIds=[], region) {
+  const figure = document.querySelector(`#${figureId}`)
+  const canvasPanel = figure.querySelector('canvas-panel')
 
   /**
-   * Handle annotation/choice selection with canvasPanel API
-   * {@link https://iiif-canvas-panel.netlify.app/docs/examples/handling-choice}
+   * Update Canvas state
    */
-  for (const canvasPanel of canvasPanels) {
-    const figure = canvasPanel.closest('.q-figure, q-lightbox')
-    if (!figure) return
-    const inputs = figure.querySelectorAll('.annotations-ui .annotations-ui__input')
-    for (const input of inputs) {
-      const annotationType = input.getAttribute('data-annotation-type')
-      const id = input.getAttribute('id')
-      const inputType = input.getAttribute('type')
-      switch (annotationType) {
-        case 'choice':
-          const toggleChoice = ({ checked }) => {
-            canvasPanel.makeChoice(id, {
-              deselect: !checked,
-              deselectOthers: inputType === 'radio'
-            })
-            if (inputType !== 'checkbox') return
-            const checkedInputs = figure.querySelectorAll('.annotations-ui__input[checked]')
-            if (!checked && checkedInputs.length === 1) {
-              checkedInputs[0].setAttribute('disabled', true)
-            }
-            if (checked) {
-              const disabledInput = figure.querySelector('[disabled]')
-              disabledInput ? disabledInput.removeAttribute('disabled') : null
-            }
-          }
-          toggleChoice(input)
-          input.addEventListener('click', ({ target }) => {
-            toggleChoice(target)
-          })
-          break
-        case 'annotation':
-          const toggleAnnotation = ({ checked }) => {
-            canvasPanel.applyStyles(id, { opacity: Number(!!checked) })
-          }
-          toggleAnnotation(input)
-          input.addEventListener('click', ({ target }) => toggleAnnotation(target));
-          break
-        default:
-          console.warn(`Invalid annotation type ${annotationType}`)
-          break
-      }
+  if (region) canvasPanel.setAttribute('region', region)x
```

Comment:
Is the `x` at the end of the statement a typo?
```suggestion
  if (region) canvasPanel.setAttribute('region', region)
```